MIP/MIP-56/README.md

```diff
@@ -0,0 +1,68 @@
+
+The Native Bridge Rate Limiter is implemented through a contract on L1. This contract is governed by the Aptos governance framework, see [MIP-48](https://github.com/movementlabsxyz/MIP/pull/48/), and can be updated through a governance process.
+
+## Motivation
+
+The correct operation of the Native Bridge relies on liveness and safety assumptions for the relayer, see [MIP-13](https://github.com/movementlabsxyz/MIP/tree/mip/security_falliblity/MIP/mip-46). If any of these would not hold, the Native Bridge could expose the network to double-spends.
+
+## Specification
+
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+![Overview](overview.png)
+*Figure 1: Overview of the Rate-Limiter mechanism for the Native Bridge.*
+
+> [!NOTE] In the following we note that when talk about the Rate-Limiter, we mean the Rate-Limiter contract.
+
+1. The Rate-Limiter SHOULD be implemented as a contract on L1.
+1. The Rate-Limiter MUST handle both the **transfer directions** L1 -> L2 and L2 -> L1. 
+1. For either transfer direction the transfer value MUST be tracked, i.e. the transferred value for L1->L2 should be recored in `budget_L1L2` and for L2->L1 in `budget_L2L1`.
+1. The Rate-Limiter SHOULD be governed by the Aptos governance framework, see [MIP-48](https://github.com/movementlabsxyz/MIP/pull/48/), and can be updated through a governance process.
+1. The security fund contract, see [MIP-50](https://github.com/movementlabsxyz/MIP/pull/50), MUST update `security_fund` in the Rate-Limiter contract if the amount of funds changes in the security fund contract.
+1. The `risk_period` value SHOULD be set by the governance process and updated through the governance process. It estimates the maximum reaction time to check whether transfers have been completed correctly.
+1. For a given transfer direction, the Rate-Limiter should disable transfers across the bridge if the rate limit is exceeded for that transfer direction.
+1. The rate limit MUST be set according to the equation `rate_limit = security_fund / risk_period * 0.5`. This is to ensure that each bridge transfer direction is ensured sufficiently.
+1. The Rate-Limiter MUST reject a transfer request if the rate limit is exceeded for that transfer direction by that transaction. I.e. for L1->L2 the Rate-Limiter MUST reject any transfer that would violate `budget_L1L2 < rate_limit` and for L2->L1 the Rate-Limiter MUST reject any transfer that would violate `budget_L2L1 < rate_limit`.
```

Comment:
To drill down a bit. The case :
L1->L2 is fine, as it will just internally block `initiateBridgeTransfer`
Whereas 
L2->L1 the only contract L1 has visibility of in this direction is the `AtomicBridgeCounteropartyMOVE.sol`, so the only call it can block is the `lockBridgeTransfer` call. 

Is this what you intend?

Reply:
it seems the function that would apply the rate limit should be `lock_bridge_transfer` on L1 according to Figure 2 in https://github.com/movementlabsxyz/MIP/pull/39

i assume currently a failed lock on L1 would not be reported to L2? Or is there a way the user on L2 can learn that the transfer request failed due to rate limit on L1?

Reply:
also see the next code line that i just added (L35 currently: ... The rate limiation should be applied at the earliest ....)

Reply:
Why we don't apply rater limiting on both side to the initiate_transfer. There's no need to have it only on one chain. They just have to share the same equation.

Reply:
Would also be possible, but requires two pools : one security pool on L1 and one security pool on L2. 

Downsides: 
- `risk_period` has to be set twice (once on L1 and once on L2)
- have to implement rate limiter on solidity and on Move
- need two security fund contracts, once in solidity and once in Move

Reply:
added it as an optimization suggestion

Reply:
With the addition of two pools this adds too much complexity for something Minimal that achieves that rate-limit guarantee that we need. I am going to implement on L1 only if we want to add Rate Limiter on the L2 later, this can happen post Bilbao Model. Or, can be considered as part of the Biarritz Model.